src/wallet/wallet.cpp

```diff
@@ -4034,17 +4041,26 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
 
     LOCK(cs_main);
 
+    bool subscribe_to_chain_updates = !gArgs.GetBoolArg("-disablewalletupdates", false);
     CBlockIndex *pindexRescan = chainActive.Genesis();
     if (!gArgs.GetBoolArg("-rescan", false))
     {
-        CWalletDB walletdb(*walletInstance->dbw);
-        CBlockLocator locator;
-        if (walletdb.ReadBestBlock(locator))
-            pindexRescan = FindForkInGlobalIndex(chainActive, locator);
+        if(subscribe_to_chain_updates) {
+            CWalletDB walletdb(*walletInstance->dbw);
+            CBlockLocator locator;
+            if (walletdb.ReadBestBlock(locator))
+                pindexRescan = FindForkInGlobalIndex(chainActive, locator);
```

Comment:
Might want to add braces `{}` around that to adhere to the coding style for new code and then install `clang-format` and run the https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy script to preempt whitespace nitpicking.